src/core/style/layout.scss

```diff
@@ -27,46 +27,6 @@
   flex-direction: row;
 }
 
-[layout-padding] > [flex-sm],
```

Comment:
So if I understand correctly, this block that you've removed was the original attempt at adjusting the padding/margin for responsive layouts, and it just failed miserably - what you've added below addresses it properly.